db/column_family.cc

```diff
@@ -466,7 +466,7 @@ void SuperVersion::Cleanup() {
   // decrement reference to the immutable MemtableList
   // this SV object was pointing to.
   imm->Unref(&to_delete);
-  MemTable* m = mem->Unref();
+  auto m = static_cast<MemTable*>(mem->Unref());
```

Comment:
Leaving `SuperVersion` to only handle read only Memtable seems reasonable, why casting this to `MemTable`. In fact, now that the read only interfaces are separated out, I wonder if we can make `SuperVersion` just hold a `ReadOnlyMemTable` interface for the mutable memtable `mem`.

Reply:
I did the casting since mem is of type MemTable. But it seems fine to have m as ReadOnlyMemtable*, updated.

Reply:
Could the `MemTable* mem;` member in SuperVersion be converted to `ReadOnlyMemTable* mem;`, since all the access to mem via SuperVersion should only be doing reading type of operations.

Reply:
Yes, updated.